sdk/core/azure-core/src/context.cpp

```diff
@@ -12,7 +12,7 @@ Azure::DateTime Azure::Core::Context::GetDeadline() const
   // Contexts form a tree. Here, we walk from a node all the way back to the root in order to find
   // the earliest deadline value.
   auto result = DateTime::max();
-  for (auto ptr = m_contextSharedState; ptr; ptr = ptr->Parent)
+  for (std::shared_ptr<ContextSharedState> ptr = m_contextSharedState; ptr; ptr = ptr->Parent)
```

Comment:
Question: What are your thoughts on such a change? Is `auto` not appropriate here.

Reply:
THis change was to fix a bug reported by the code analysis tools - apparently auto cannot properly deduce the type of the destination, causing extra copies. Having said that, I think the right fix is to say `auto & ptr` or `const auto& ptr`.